File: 프로그래머스/unrated/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.cs
using System;

public class Solution {
    public int solution(int[,] arr) 
    {
        for(int y = 0; y < arr.GetLength(0); ++y)
        {
            for(int x = 0; x < arr.GetLength(1); ++x)
            {
                if(arr[y, x] != arr[x, y])
                    return 0;
            }
        }
        
        return 1;
    }
}

File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> arr) {
    
    for (int y = 0; y < arr.size(); ++y)
    {
        for (int x = 0; x < y; ++x)
        {
            if (arr[y][x] != arr[x][y])
                return 0;
        }
    }
    
    return 1;
}